xdsl/frontend/python_code_check.py

```diff
@@ -45,29 +45,33 @@ def run(stmts: List[ast.stmt]) -> None:
                 break
 
         # Check Python code is correctly structured.
-        CheckStructure.run_with_scope(single_scope, stmts)
+        CheckStructure.run_with_scope(single_scope, stmts, file)
 
         # Check constant/global variables are correctly defined. Should be
         # called only after the structure is checked.
-        CheckAndInlineConstants.run(stmts)
+        CheckAndInlineConstants.run(stmts, file)
 
 
 @dataclass
 class CheckStructure:
 
     @staticmethod
-    def run_with_scope(single_scope: bool, stmts: List[ast.stmt]) -> None:
+    def run_with_scope(single_scope: bool, stmts: List[ast.stmt],
+                       file: str) -> None:
         if single_scope:
-            visitor = SingleScopeVisitor()
+            visitor = SingleScopeVisitor(file)
         else:
-            visitor = MultipleScopeVisitor()
+            visitor = MultipleScopeVisitor(file)
         for stmt in stmts:
             visitor.visit(stmt)
 
 
 @dataclass
 class SingleScopeVisitor(ast.NodeVisitor):
 
+    file: str = field(default="<unknown file>")
+    """File name for error reporting."""
```

Comment:
nit: this can be a bit confusing in the future, because it's a path right?

Reply:
Good catch, thanks! The dangers of co-pilot autocomplete 🙈